typings/index.d.ts

```diff
@@ -159,9 +161,23 @@ declare module 'mappersmith' {
     setHeaders(xmlHttpRequest: XMLHttpRequest, headers: Headers): void
   }
 
+  export type HTTPRequestOptions = http.RequestOptions | https.RequestOptions
```

Comment:
This is my best guess of what `requestParams` in https://github.com/tulios/mappersmith/blob/master/src/gateway/http.js#L82 should be typed as. Any other input or guess is appreciated! 😄